### DISCUSSION TITLE
Error messages with infix operators

### DISCUSSION THREAD

**Discussion by edahlseng** [pointfreeco/swift-prelude#161] [Q&A]
The infix operators introduced in this library are incredibly useful! They've dramatically improved the readability and expressiveness in my projects as I continue to use them more and more. I've run into some difficulties when using them, though, and hoping you could help. Often times I'll get errors when using an infix operator, but the errors are quite unhelpful. They tend to look like the following:

```
error: cannot convert value of type '(Any?) -> Preferences' to expected argument type '(_) -> _'
```

The compiler just spits out `_` instead of telling me which error type it is expecting. If I remove the infix operators and write out the problematic line the "long way," then usually the error messages are quite helpful, and I can easily find my mistake. Have you seen this before? Is there a way to fix this?

**Comment by stephencelis** [comment]
This is unfortunately the way the Swift compiler diagnoses things. It's gotten better over the years, but the best way to work around obscure errors, in our experience, is to give the compiler as much type information as possible, at least temporarily while debugging.

Going to convert this to a discussion for now. Sorry for the quite long delay :)
